impeller/renderer/render_pass.h

```diff
@@ -60,6 +63,81 @@ class RenderPass {
   ///
   bool AddCommand(Command&& command);
 
+  //----------------------------------------------------------------------------
+  /// The pipeline to use for this command.
+  void SetPipeline(
+      const std::shared_ptr<Pipeline<PipelineDescriptor>>& pipeline);
+
+  //----------------------------------------------------------------------------
+  /// The debugging label to use for the command.
+  void SetCommandLabel(const std::string& label);
+
+  //----------------------------------------------------------------------------
+  /// The reference value to use in stenciling operations. Stencil configuration
+  /// is part of pipeline setup and can be read from the pipelines descriptor.
+  ///
+  /// @see         `Pipeline`
+  /// @see         `PipelineDescriptor`
+  ///
+  void SetStencilReference(uint32_t value);
+
+  void SetBaseVertex(uint64_t value);
+
+  //----------------------------------------------------------------------------
+  /// The viewport coordinates that the rasterizer linearly maps normalized
+  /// device coordinates to.
+  /// If unset, the viewport is the size of the render target with a zero
+  /// origin, znear=0, and zfar=1.
+  ///
+  void SetViewport(Viewport viewport);
+
+  //----------------------------------------------------------------------------
+  /// The scissor rect to use for clipping writes to the render target. The
+  /// scissor rect must lie entirely within the render target.
+  /// If unset, no scissor is applied.
+  ///
+  void SetScissor(IRect scissor);
+
+  //----------------------------------------------------------------------------
+  /// The number of instances of the given set of vertices to render. Not all
+  /// backends support rendering more than one instance at a time.
+  ///
+  /// @warning      Setting this to more than one will limit the availability of
+  ///               backends to use with this command.
+  ///
+  void SetInstanceCount(size_t count);
+
+  //----------------------------------------------------------------------------
+  /// @brief      Specify the vertex and index buffer to use for this command.
+  ///
+  /// @param[in]  buffer  The vertex and index buffer definition. If possible,
+  ///             this value should be moved and not copied.
+  ///
+  /// @return     returns if the binding was updated.
+  ///
+  bool SetVertexBuffer(VertexBuffer buffer);
+
+  /// Record the currently pending command.
+  bool Dispatch();
```

Comment:
Please use fml::Status instead of using bool.

Reply:
This is a bool all the way up the stack, I could change this to status but then I'd convert from fml::status to bool in each contents or complete the migration.

I like the idea of fml::status in general, but I'm not sure if this is the right usage for it. Right now Draw() essentially error checks for two conditions:

1. If you set a scissor (which we never do) is it a valid scissor?
2. Is the pipeline object defined before draw was called?

I think the first check should be moved into the setter, but really that is something we should VALIDATION_LOG and not worry about. The second one seems like it should be an FML_DCHECK.

Reply:
That's fine if this gets converted to bool in the next level up.  I have an issue out to convert all of these bools over to fml::Status.  We need to get these all migrated at some point `bool` is ambiguous.  So even at just this one layer it's worthwhile.  Once everything is using it and StatusOr, magic.

Reply:
Done - i'll be able to make the message better once I finish migrating the API since I can just use the validation message.